src/workerd/io/worker-interface.c++

```diff
@@ -222,8 +222,12 @@ kj::Promise<void> RevocableWorkerInterface::connect(kj::StringPtr host, const kj
 }
 
 RevocableWorkerInterface::RevocableWorkerInterface(WorkerInterface& worker,
-    kj::Promise<void> revokeProm)
-    : worker(worker), revokeProm(revokeProm.fork()) {}
+    kj::Promise<void> revoker)
+    : worker(worker), revokeProm(revoker.fork()),
+        cancelProm(revokeProm.addBranch().catch_([this](kj::Exception&& e) -> kj::Promise<void> {
+          canceler.cancel(kj::cp(e));
+          return kj::READY_NOW;
+        }).eagerlyEvaluate(nullptr)) {}
```

Comment:
Perhaps instead of using the lambda here, define a utility method as a coroutine... e.g. something like...

```cpp
kj::Promise<void> handleRevokePromise(kj::Promise<void> promise) {
  try {
    co_await promise;
  } catch (kj::Exception& ex) {
    canceler.cancel(kj::cp(ex));
  }
}

// ...
  //....
  cancelProm(handleRevokePromise(revokeProm.addBranch()).eagerlyEvaluate()) { // ...
```